src/theory/datatypes/datatypes_rewriter.cpp

```diff
@@ -76,8 +76,7 @@ Node DatatypesRewriter::rewriteViaRule(ProofRewriteRule id, const Node& n)
       Assert(tn.isDatatype());
       const DType& dt = tn.getDType();
       size_t i = utils::indexOf(n.getOperator());
-      bool sharedSel = d_opts.datatypes.dtSharedSelectors;
-      Node ticons = utils::getInstCons(t, dt, i, sharedSel);
+      Node ticons = utils::getInstCons(t, dt, i, false);
```

Comment:
I'm a bit confused by the intended behavior here. IIUC one can still use shared selectors and proofs, but in that case this rule would not reflect that shared selectors are being used? How would that work?

Reply:
In this case, the "official" proof rule would not match what the rewriter did, so RARE proof reconstruction would fail.

Reply:
I see. Can you please add a comment to clarify this?